miio/cooker.py

```diff
@@ -0,0 +1,706 @@
+import enum
+import logging
+import string
+from collections import defaultdict
+from datetime import time
+from typing import Optional
+
+import click
+
+from .click_common import command, format_output, EnumType
+from .device import Device, DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_PRESSURE1 = 'chunmi.cooker.press1'
+MODEL_PRESSURE2 = 'chunmi.cooker.press2'
+MODEL_NORMAL1 = 'chunmi.cooker.normal1'
+MODEL_NORMAL2 = 'chunmi.cooker.normal2'
+MODEL_NORMAL4 = 'chunmi.cooker.normal3'
+MODEL_NORMAL3 = 'chunmi.cooker.normal4'
+MODEL_NORMAL5 = 'chunmi.cooker.normal5'
+
+MODEL_PRESSURE = [MODEL_PRESSURE1, MODEL_PRESSURE2]
+MODEL_NORMAL = [MODEL_NORMAL1, MODEL_NORMAL2, MODEL_NORMAL3, MODEL_NORMAL4,
+                MODEL_NORMAL5]
+
+MODEL_NORMAL_GROUP1 = [MODEL_NORMAL2, MODEL_NORMAL5]
+MODEL_NRRMAL_GROUP2 = [MODEL_NORMAL3, MODEL_NORMAL4]
+
+COOKING_STAGE_NAME = [
+    'Preheating',
+    'Water-absorbing',
+    'Boiling',
+    'Gelantinizing',
+    'Braising',
+]
+
+COOKING_STAGE_DESCRIPTION = [
+    'Starting heating, to soften rice gradually',
+    'Increase temperature, to flesh grains with water',
+    'Last high heating, to cook rice evenly',
+    'Steaming under high temperature, to bring sweetness to grains',
+    'Reheat the rice to open its taste',
+]
+
+
+class CookerException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    # Observed
+    Running = 'running'
+    Waiting = 'waiting'
+    AutoKeepWarm = 'autokeepwarm'
+    # Potential candidates
+    Cooking = 'cooking'
+    Finish = 'finish'
+    FinishA = 'finisha'
+    KeepWarm = 'keepwarm'
+    KeepTemp = 'keep_temp'
+    Notice = 'notice'
+    Offline = 'offline'
+    Online = 'online'
+    PreCook = 'precook'
+    Resume = 'resume'
+    ResumeP = 'resumep'
+    Start = 'start'
+    StartP = 'startp'
+    Cancel = 'Отмена'
+
+
+class TemperatureHistory:
```

Comment:
doctstrings would be nice for this class, too. What does this dow and how can I use it? It's a container of temperatures recorded every X time interval? Is there a limit how many historical values are available?